### PR TITLE
fit redux-saga 0.9.1

### PR DESCRIPTION
**Pull request by dimitrikochnev**
v0.9.1
- `select(selector, ...args)`

v0.8.0
- `takeEvery` / `takeLatest`
- `isCancelError`
- `import {  take, ... } from 'redux-saga'` -> `import {  take, ... } from 'redux-saga/effects`

v0.7.0:
- `task.cancel()` 
- `sagaMiddleware.run(dynamicSaga)`

### CODE REVIEW

**Comment by aikoven** [comment]
I'll review it tomorrow.
Сб, 27 февр. 2016 г. в 21:10, Definitely Bot <notifications@github.com>:

**Comment by aikoven** [comment]
Everything looks good except for missing `redux-saga/utils` module and one export from the main module:

https://github.com/yelouafi/redux-saga/blob/master/src/index.js
```ts
import * as effects from './effects'
import * as utils from './utils'

export {
  effects,
  utils
}
```

**Comment by aikoven** [comment]
Also, `apply` effect creator is missing.

**Comment by rosendi** [comment]
@aikoven I'm not sure about your first comment. I can't find it in the public API doc. Typings should cover   API that is mentioned in docs, not source code. 

The `apply` effect has been added. Please, check the latest one commit and if it passes, I'll rebase it.

File: redux-saga/redux-saga.d.ts [hunk 1]
```diff
@@ -9,56 +9,83 @@ declare module 'redux-saga' {
...
-    RACE_AUTO_CANCEL,
-    PARALLEL_AUTO_CANCEL,
-    MANUAL_CANCEL
-  } from 'redux-saga/lib/proc';
+  type EffectFunction<T1, T2, T3> = (arg1?: T1, arg2?: T2, arg3?: T3, ...rest: any[]) => Promise<any> | Iterable<any>;
 
-  import * as monitorActions from 'redux-saga/lib/monitorActions';
-  export  {monitorActions}
+  interface EffectFunctionContext<T1, T2, T3> {
+    0: Object;
```
Comment (rosendi) [inline redux-saga/redux-saga.d.ts]: 0: Object; or 0: any; ?

File: redux-saga/redux-saga.d.ts [hunk 2]
```diff
@@ -9,56 +9,83 @@ declare module 'redux-saga' {
...
-}
+  export function put(action: any): Effect;
+
+  export function race(effects: {[key:string]: any}): Effect;
 
+  export function call<T1, T2, T3>(fn: EffectFunction<T1, T2, T3>, arg1?: T1, arg2?: T2, arg3?: T3, ...rest: any[]): Effect;
+
+  export function call<T1, T2, T3>(fn: EffectFunctionContext<T1, T2, T3>, arg1?: T1, arg2?: T2, arg3?: T3, ...rest: any[]): Effect;
+
+  export function apply<T1, T2, T3>(context: Object, fn: EffectFunction<T1, T2, T3>, arg1?: T1, arg2?: T2, arg3?: T3, ...rest: any[]): Effect;
```
Comment (rosendi) [inline redux-saga/redux-saga.d.ts]: same here: 0: Object; or 0: any; ?

**Comment by aikoven** [comment]
👍

**Pull request merged**